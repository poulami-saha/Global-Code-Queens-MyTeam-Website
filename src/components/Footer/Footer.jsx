import React from "react";
import Typography from "../utility/Typography/Typography";
import styles from "./Footer.module.css";
import logo from "../../assets/logo.svg";
import facebook from "../../assets/icon-facebook.svg";
import pinterest from "../../assets/icon-pinterest.svg";
import twitter from "../../assets/icon-twitter.svg";
import { Link } from "react-router-dom";

function Footer() {
  const today = new Date();
  const thisYear = today.getFullYear();

  return (
    <footer className={styles.footer}>
      <div className={styles.footerInfo}>
        <div className={styles.footerNavigation}>
          <img src={logo} alt="logo" />
          <div className={styles.footerLinks}>
            <Link to="/" className={styles.link}>
              <Typography elType="body1">home</Typography>
            </Link>
            <Link to="/about" className={styles.link}>
              <Typography elType="body1">about</Typography>
            </Link>
          </div>
        </div>

        <div className={styles.location}>
          <Typography elType="body1">997 Hillcrest Lane</Typography>
          <Typography elType="body1">Irvine, CA</Typography>
          <Typography elType="body1">California 92714</Typography>
          <Typography elType="body1">Call Us: 949-833-7432</Typography>
        </div>
      </div>

      <div className={styles.socialContent}>
        <div className={styles.socialLinks}>
          <img src={facebook} alt="facebook" />
          <img src={pinterest} alt="pinterest" />
          <img src={twitter} alt="X" />
        </div>

        <Typography elType="body1" className={styles.copyright}>
          Copyright {thisYear}. All Rights Reserved
        </Typography>
      </div>
    </footer>
  );
}

export default Footer;
