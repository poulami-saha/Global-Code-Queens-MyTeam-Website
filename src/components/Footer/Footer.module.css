


.footerContent {
    background-color: #012F34;
    position: relative;
    padding: 20px;
    transition: background-color 0.3s, color 0.3s;
  }
  

  
  
  .footerInfo {
    display: flex;
    justify-content: space-between;
    padding: 10px 20px;
    align-items: center;
  }
  
  .logoDesc {
    display: flex;
    flex-direction: column;
  }
  
  .footerLinks {
    display: flex;
    gap: 1rem;
    font-size: 0.9rem;
    margin-top: 0.9rem;
    /* flex-direction: column; */
  }
  
  .link {
    color: #FFFFFF;
    text-decoration: none;
    margin: 5px 0;
  }
  
  .location {
    color: #979797;
    font-size: 1rem;
  }
  
  .socialContent {
    display: flex;
    flex-direction: column;
    align-items: center;
    
  }
  
  .socialLinks {
    display: flex;
    gap: 1rem;
    cursor: pointer;
    
  }

  .socialLinks:hover {
    color: #012F34;
    background-color: aqua;
  }


  
  .copyright {
    color: #979797;
    /* color: inherit; */
    font-size: 0.9rem;
    margin-top: 2.5rem;
    align-items: center;
  }
  
 
  
 

  @media (max-width: 768px) {
    .footerContent {
        padding: 70px ;
    }
    
    .footerInfo {
      flex-direction: row;
      align-items: center;
      text-align: left;
      position: relative;
    }
  
    .footerLinks {
      flex-direction: row;
    }

    .socialLinks {
        position: absolute;
        left: 24px;
        bottom: -40px;
        
    }

    .location {
        position: absolute;
        right: -30px;
        bottom: -30px;
        text-align: end;
        
    }


    .copyright {
        position: absolute;
        bottom: -65px;
        right: -30px;


    }
  
    .socialContent {
      align-items: flex-end;

    }
  
    .themeToggleContainer {
      top: 5px;
      right: -10px;
    }
  
    .themeToggleText {
      margin-left: 10px;
    }
  }
  
  @media (min-width: 960px) {
    .footerInfo {
      flex-direction: row;
      align-items: center;
      text-align: left;
    }
  
    .footerLinks {
      flex-direction: row;
    }


  
    .socialContent {
      align-items: flex-end;
    }
  
    .themeToggleContainer {
      top: 20px;
      right: 20px;
    }
  
    .themeToggleText {
      margin-left: 10px;
    }

    
  }





 
  