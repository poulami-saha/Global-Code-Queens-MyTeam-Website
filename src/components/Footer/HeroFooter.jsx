import React from "react";
import { useEffect, useState } from "react";
import Typography from "../utility/Typography/Typography";
import styles from "../Footer/HeroFooter.module.css";
import logo from "../../assets/logo.svg";
import facebook from "../../assets/icon-facebook.svg";
import pinterest from "../../assets/icon-pinterest.svg";
import twitter from "../../assets/icon-twitter.svg";
import toggleButton from "../../assets/on-button-removebg-preview.png";
import { Link } from "react-router-dom";

function HeroFooter() {
  const [theme, setTheme] = useState(
    localStorage.getItem("footerTheme") || "light"
  );

  useEffect(() => {
    localStorage.setItem("footerTheme", theme);
  }, [theme]);

  const toggleTheme = () => {
    setTheme((prevTheme) => (prevTheme === "light" ? "dark" : "light"));
  };

  return (
    <div className={`${styles.footerContent} ${styles[theme]}`}>
      <div className={styles.themeToggleContainer} style={{ transform: theme === 'light' ? 'translateX(0)' : 'translateX(-50px)' }}>
        
        <img
          src={toggleButton}
          alt="light/dark"
          onClick={toggleTheme}
          className={styles.themeToggle}
        />
        <Typography elType="body1" className={styles.themeToggleText}>
          {/* {theme === 'dark' ? 'Dark Mode' : 'Light Mode'} */}
        </Typography>
      </div>

      <div className={styles.footerInfo}>
        <div className={styles.logoDesc}>
          <img src={logo} alt="logo" />
          <div className={styles.footerLinks}>
            <Link to="/" className={styles.link}>
              <Typography elType="body1">home</Typography>
            </Link>
            <Link to="/about" className={styles.link}>
              <Typography elType="body1">about</Typography>
            </Link>
          </div>
        </div>

        <div className={styles.location}>
          <Typography elType="body1">997 Hillcrest Lane</Typography>
          <Typography elType="body1">Irvine, CA</Typography>
          <Typography elType="body1">California 92714</Typography>
          <Typography elType="body1">Call Us: 949-833-7432</Typography>
        </div>

        <div className={styles.socialContent}>
          <div className={styles.socialLinks}>
            <img src={facebook} alt="facebook" />
            <img src={pinterest} alt="pinterest" />
            <img src={twitter} alt="X" />
          </div>

          <Typography elType="body1" className={styles.copyright}>
            Copyright 2020. All Rights Reserved
          </Typography>
        </div>
      </div>
    </div>
  );
}

export default HeroFooter;
