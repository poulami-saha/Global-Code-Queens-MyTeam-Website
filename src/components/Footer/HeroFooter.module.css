
.footerContent {
    background-color: #012F34;
    position: relative;
    padding: 20px;
    transition: background-color 0.3s, color 0.3s;
  }
  
  .footerContent.light {
    background-color: #ffffff;
    color: #012f34;
  }
  
  .footerContent.dark {
    background-color: #012f34;
    color: #ffffff;
  }
  
  .footerInfo {
    display: flex;
    justify-content: space-between;
    padding: 10px 20px;
    align-items: center;
  }
  
  .logoDesc {
    display: flex;
    flex-direction: column;
  }
  
  .footerLinks {
    display: flex;
    gap: 1rem;
    font-size: 0.9rem;
    margin-top: 0.9rem;
    /* flex-direction: column; */
  }
  
  .link {
    text-decoration: none;
    color: inherit;
    margin: 5px 0;
  }
  
  .location {
    color: #979797;
    font-size: 1rem;
  }
  
  .socialContent {
    display: flex;
    flex-direction: column;
    align-items: center;
    
  }
  
  .socialLinks {
    display: flex;
    gap: 1rem;
    
  }


  
  .copyright {
    color: #979797;
    /* color: inherit; */
    font-size: 0.9rem;
    margin-top: 2.5rem;
    align-items: center;
  }
  
  .themeToggleContainer {
    position: absolute;
    top: 10px;
    right: 20px;
    display: flex;
    align-items: center;
    transition: transform 0.3s ease, right 0.3s ease; }
  
  .themeToggle {
    cursor: pointer;
    width: 40px;
    height: 40px;
  }
  
  .themeToggleText {
    margin-left: 10px;
    cursor: pointer;
  }
  
  .themeToggle:hover,
  .themeToggleText:hover {
    opacity: 0.8;
  }

  @media (max-width: 768px) {
    .footerContent {
        padding: 70px ;
    }
    
    .footerInfo {
      flex-direction: row;
      align-items: center;
      text-align: left;
      position: relative;
    }
  
    .footerLinks {
      flex-direction: row;
    }

    .socialLinks {
        position: absolute;
        left: 24px;
        bottom: -40px;
        
    }

    .location {
        position: absolute;
        right: -30px;
        bottom: -30px;
        text-align: end;
        
    }


    .copyright {
        position: absolute;
        bottom: -65px;
        right: -30px;


    }
  
    .socialContent {
      align-items: flex-end;

    }
  
    .themeToggleContainer {
      top: 5px;
      right: -10px;
    }
  
    .themeToggleText {
      margin-left: 10px;
    }
  }
  
  @media (min-width: 960px) {
    .footerInfo {
      flex-direction: row;
      align-items: center;
      text-align: left;
    }
  
    .footerLinks {
      flex-direction: row;
    }


  
    .socialContent {
      align-items: flex-end;
    }
  
    .themeToggleContainer {
      top: 20px;
      right: 20px;
    }
  
    .themeToggleText {
      margin-left: 10px;
    }

    
  }




/* old styles */
/* .footerContent {
  background-color: #012F34;
  position: relative;
  padding: 20px;
  transition: background-color 0.3s, color 0.3s;
}

.footerInfo {
  position: relative;
  display: flex;
  justify-content: space-around;
  padding: 10px 20px;
  align-items: center;
}

.location {
  color: #979797;
  font-size: 0.9rem;
}

.link {
  text-decoration: none;
  color: #ffffff;
  margin: 5px 0;
}

.footerLinks {
  display: flex;
  gap: 1rem;
  font-size: 0.9rem;
  cursor: pointer;
  text-decoration: none;
}

.socialLinks {
  display: flex;
  gap: 1rem;
  justify-content: center;
  cursor: pointer;
}

.socialContent {
  display: flex;
  justify-content: right;
  flex-direction: column;
}

.copyright {
  color: #979797;
  font-size: 0.9rem;
}

.footerContent.light {
  background-color: #ffffff;
  color: #012f34;
}

.footerContent.dark {
  background-color: #012f34;
  color: #ffffff;
}

.footerInfo {
  display: flex;
  justify-content: space-between;
}

.themeToggle {
    margin-top: 20px;
    cursor: pointer;
    border: none;
    width: 50px;
    transition: background-color 0.5s;
    place-content: left;
  }

.themeToggleContainer {
  display: flex;
  align-items: center;
  position: absolute;
  top: 20px;
  right: 20px;
}

.themeToggle {
  cursor: pointer;
  width: 30px;
  height: 30px;
}

.themeToggleText {
  margin-left: 10px;
  cursor: pointer;
}

.themeToggle:hover {
  opacity: 0.8;
}

.themeToggleText:hover {
  opacity: 0.8;
} */
