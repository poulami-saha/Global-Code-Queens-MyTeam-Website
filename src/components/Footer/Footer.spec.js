import React from "react";
import { render } from "@testing-library/react";
import Footer from "./Footer";

jest.mock("react-router-dom", () => {
  return {
    Link: ({ to, children }) => <a href={to}>{children}</a>,
  };
});

test("renders NotFound correctly and matches snapshot", async () => {
  const { asFragment, getByText } = render(<Footer />);

  // Snapshot test
  expect(asFragment()).toMatchSnapshot();

  // Content test
  expect(getByText(/Copyright 2020./)).toBeInTheDocument();
  expect(getByText(/All Rights Reserved/)).toBeInTheDocument();
  expect(getByText(/997 Hillcrest Lane/)).toBeInTheDocument();
  expect(getByText(/Irvine, CA/)).toBeInTheDocument();
  expect(getByText(/California 92714/)).toBeInTheDocument();
  expect(getByText(/Call Us: 949-833-7432/)).toBeInTheDocument();
});
