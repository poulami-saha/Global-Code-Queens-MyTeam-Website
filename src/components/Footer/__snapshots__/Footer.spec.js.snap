// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NotFound correctly and matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="footerInfo"
    >
      <div
        class="footerNavigation"
      >
        <img
          alt="logo"
          src="logo.svg"
        />
        <div
          class="footerLinks"
        >
          <a
            href="/"
          >
            <p
              class="body1"
              id=""
            >
              home
            </p>
          </a>
          <a
            href="/about"
          >
            <p
              class="body1"
              id=""
            >
              about
            </p>
          </a>
        </div>
      </div>
      <address
        class="location"
      >
        <p
          class="body2"
          id=""
        >
          997 Hillcrest Lane
        </p>
        <p
          class="body2"
          id=""
        >
          Irvine, CA
        </p>
        <p
          class="body2"
          id=""
        >
          California 92714
        </p>
        <p
          class="body2"
          id=""
        >
          Call Us: 949-833-7432
        </p>
      </address>
    </div>
    <div
      class="socialContent"
    >
      <div
        class="socialLinks"
      >
        <svg
          class="socialLink"
        >
          icon-facebook.svg
        </svg>
        <svg
          class="socialLink"
        >
          icon-pinterest.svg
        </svg>
        <svg
          class="socialLink"
        >
          icon-twitter.svg
        </svg>
      </div>
      <p
        class="copyright body2"
        id=""
      >
        <span
          class="avoidWrap"
        >
          Copyright 2020.
        </span>
         
        <span
          class="avoidWrap"
        >
          All Rights Reserved
        </span>
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
